File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        Deque<int[]> queue = new ArrayDeque<>();
        for (int i = 0; i < prices.length; i++) {
            queue.offer(new int[] { i, prices[i] });
        }
        int[] answer = new int[prices.length];

        while (!queue.isEmpty()) {
            int[] current = queue.poll();
            int time = 0;
            for (int[] p : queue) {
                time++;
                if (p[1] < current[1]) {
                    break;
                }
            }
            answer[current[0]] = time;
        }
        return answer;
    }
}

File: 프로그래머스/2/42584. 주식가격/주식가격.py
from collections import deque
def solution(prices):
    queue = deque(enumerate(prices))
    answer = []
    # [1,2,3,2,3]
    # [4,3,1,1,0]
    while queue:
        now_idx, now_value = queue.popleft()
        count = 0
        for idx, value in queue:
            count += 1
            if now_value > value:
                break
        answer.append(count)
    return answer